piccolo_api/crud/hooks.py

```diff
@@ -0,0 +1,45 @@
+from enum import Enum
+import typing as t
+
+from piccolo.table import Table
+
+
+class HookType(Enum):
+    pre_save = "pre_save"
+    pre_patch = "pre_patch"
+    pre_delete = "pre_delete"
+
+
+class Hook:
+    def __init__(self, hook_type: HookType, coro) -> None:
```

Comment:
We might want to rename this from `coro` to `callable`. We can then have this as the type annotation:

```python
class Hook:
    def __init__(self, hook_type: HookType, callable: t.Callable) -> None:
```

The problem with coro, is we're asking them to provide an async function, not an actual coroutine.

It also means we can support async and non-async functions. Wherever we run the hook, we can just do this:

```python
if inspect.iscoroutinefunction(hook.callable):
    await hook.callable()
else:
    hook.callable()
```

Reply:
makes total sense. Will fix.